xds/src/main/java/io/grpc/xds/ClusterMetadataRegistry.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2024 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.xds;
+
+import com.google.protobuf.Any;
+import com.google.protobuf.InvalidProtocolBufferException;
+import io.envoyproxy.envoy.extensions.filters.http.gcp_authn.v3.Audience;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Registry for parsing cluster metadata values.
+ *
+ * <p>This class maintains a mapping of type URLs to {@link ClusterMetadataValueParser} instances,
+ * allowing for the parsing of different metadata types.
+ */
+final class ClusterMetadataRegistry {
+  private static final ClusterMetadataRegistry INSTANCE = new ClusterMetadataRegistry();
+
+  private final Map<String, ClusterMetadataValueParser> supportedParsers = new HashMap<>();
+
+  private ClusterMetadataRegistry() {
+    registerParsers(
```

Comment:
Make this a call to a `registerParser(String, ClusterMetadataValueParser)` or just `supportedParsers.put()` directly? The array here doesn't look to help at all and almost the entire implementation of `registerParsers()` is _undoing_ the limitations of the array.

Although if `AudienceMetadataParser` moves to gcp authn filter, then it could make sense to add a method to ClusterMetadataValueParser to get the metadata type string. Then you wouldn't need to pack the arguments into an array.